File: Javascript/Longest_palindromic_Substring.js
function longestPalindrome(s) {
    if (s === null || s.length < 1) return '';

    let start = 0, end = 0;
    for (let i = 0; i < s.length; i++) {
        let len1 = expandAroundCenter(s, i, i);
        let len2 = expandAroundCenter(s, i, i + 1);
        let len = Math.max(len1, len2);
        if (len > end - start) {
            start = i - Math.floor((len - 1) / 2);
            end = i + Math.floor(len / 2);
        }
    }
    return s.substring(start, end + 1);
}

function expandAroundCenter(s, left, right) {
    let L = left, R = right;
    while (L >= 0 && R < s.length && s.charAt(L) === s.charAt(R)) {
        L--;
        R++;
    }
    return R - L - 1;
}

let s = 'babad';
console.log(`Longest palindromic substring of ${s}: ${longestPalindrome(s)}`);

File: C#/longest_palindromic_substring.cs
using System;

class LongestPalindromicSubstring
{
    public string LongestPalindrome(string s)
    {
        if (s == null || s.Length < 1) return "";
        
        int start = 0, end = 0;
        for (int i = 0; i < s.Length; i++)
        {
            int len1 = ExpandAroundCenter(s, i, i);
            int len2 = ExpandAroundCenter(s, i, i + 1);
            int len = Math.Max(len1, len2);
            if (len > end - start)
            {
                start = i - (len - 1) / 2;
                end = i + len / 2;
            }
        }
        
        return s.Substring(start, end - start + 1);
    }
    
    private int ExpandAroundCenter(string s, int left, int right)
    {
        int L = left, R = right;
        while (L >= 0 && R < s.Length && s[L] == s[R])
        {
            L--;
            R++;
        }
        return R - L - 1;
    }

    public static void Main()
    {
        LongestPalindromicSubstring lps = new LongestPalindromicSubstring();
        string s = "babad";
        Console.WriteLine("Longest palindromic substring of " + s + ": " + lps.LongestPalindrome(s));
    }
}
